modules/control/common/mrac_controller.cc

```diff
@@ -218,16 +219,38 @@ void MracController::BuildAdaptionModel() {
     matrix_b_adaption_(1, 0) = 1.0;
     matrix_p_adaption_(0, 0) = 1.0;
     matrix_p_adaption_(1, 1) = 1.0;
-    // TODO(Yu): check whether the high-order P matrix is positive definite
   } else {
     AWARN << "Adaption model order beyond the designed range, "
              "model_order: "
           << model_order_;
     adaption_model_enabled_ = false;
   }
+  if (matrix_a_reference_.rows() != matrix_p_adaption_.rows() ||
+      matrix_a_reference_.cols() != matrix_p_adaption_.cols()) {
+    AWARN << "Adaption matrix size does not match the reference matrix size";
+    adaption_model_enabled_ = false;
+  }
+  if (!CheckLyapunovPD(matrix_a_reference_, matrix_p_adaption_)) {
+    AWARN << "Solution of the algebraic Lyapunov equation is not symmetric "
+             "positive definite";
+    adaption_model_enabled_ = false;
+  }
   adaption_model_enabled_ = true;
```

Comment:
Double check the logic here. This might not be valid due to the logic changes above.

Reply:
Fixed per comment. Using conf parameter together with the common::status::ErrorCode to control the potential config parametric error